ecommerce/core/models.py

```
@@ -45,6 +55,63 @@ class SiteConfiguration(models.Model):
     class Meta(object):
         unique_together = ('site', 'partner')
 
+    @property
+    def payment_processors_set(self):
+        """
+        Returns a set of enabled payment processor keys
+        Returns:
+            set[string]: Returns a set of enabled payment processor keys
+        """
+        return {
+            raw_processor_value.strip()
+            for raw_processor_value in self.payment_processors.split(',')
+        }
+
+    def _clean_payment_processors(self):
+        """
+        Validates payment_processors field value
+
+        Raises:
+            ValidationError: If `payment_processors` field contains invalid/unknown payment_processor names
+        """
+        value = self.payment_processors.strip()
+        if not value:
+            raise ValidationError('Invalid payment processors field: must not only contain whitespace characters')
+
+        processor_names = value.split(',')
+        for name in processor_names:
+            try:
+                get_processor_class_by_name(name.strip())
+            except ProcessorNotFoundError as exc:
+                raise ValidationError(exc.message)
+
+    def get_payment_processors(self):
+        """
+        Returns payment processor classes enabled for the corresponding Site
+
+        Returns:
+            list[BasePaymentProcessor]: Returns payment processor classes enabled for the corresponding Site
+        """
+        all_processors = [get_processor_class(path) for path in settings.PAYMENT_PROCESSORS]
+        all_processor_names = {processor.NAME for processor in all_processors}
+
+        missing_processor_configurations = self.payment_processors_set - all_processor_names
+        if missing_processor_configurations:
+            processor_config_repr = ", ".join(missing_processor_configurations)
+            log.warning(
```

If you're using this for startup validation, logging a warning won't be good enough. You need to raise an error to actually stop execution.